### PR TITLE
RWS 2.0 API Update

### PR DESCRIPTION
**Pull request by JcJakeway**
What:
Updates our existing function blocks to work with RWS 2.0 API.

Why:
All new ABB OmniCore robots use this API.

### CODE REVIEW

File: example/AsProject/Logical/Libraries/Loupe/Package.pkg [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
+    <Object Type="Library" Language="Binary">tcpcomm</Object>
     <Object Type="Library" Language="Binary">ringbuflib</Object>
     <Object Type="Library" Language="Binary">stringext</Object>
-    <Object Type="Library" Language="Binary">jsmn_parse</Object>
-    <Object Type="Library" Language="Binary">tcpcomm</Object>
-    <Object Type="Library" Language="ANSIC" Reference="true">..\..\src\Ar\A3BR\ANSIC.lby</Object>
   </Objects>
-</Package>
\ No newline at end of file
+</Package>
```
[the remark this answers is not in the document]
Comment (JcJakeway) [reply example/AsProject/Logical/Libraries/Loupe/Package.pkg:11]: Noted, thanks Scott.

File: example/AsProject/Logical/ProcessControl/RobotRwsIntf/Main.st [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	rws.connect.enable := TRUE;
+	rws.connect.configuration.hostname := '192.168.125.1';  //specific per robot used
+	rws.connect.configuration.username := 'Default User';//specific per robot used
+	rws.connect.configuration.password := 'robotics';  //specific per robot used
+	rws.connect.configuration.apiVersion := A3BR_API_VERSION_2;  //Set to Version RWS 2.0 if Robot has a Omnicore processor
+	IF rws.connect.configuration.apiVersion = A3BR_API_VERSION_1 THEN
+		rws.connect.configuration.port := 80;
+	ELSE
+		rws.connect.configuration.port := 443;
+	END_IF
```
Comment (AndrewMusser) [inline example/AsProject/Logical/ProcessControl/RobotRwsIntf/Main.st:19]: Nice addition here. Can we point out in a comment that V2 uses HTTPS by default, and that's why the port needs to default to 443?

File: src/Ar/A3BR/a3brControl.c [hunk 3]
```diff
@@ -204,12 +216,17 @@ void A3brControl(struct A3brControl* inst){
...
+				brsstrcat( request.uri, "?json=1");	
 				break;
 		}
 		request.dataType = A3BR_REQ_DATA_TYPE_PARS;
 		request.errorCallback = &A3brControlErrorCallback;
 		request.successCallback = &A3brControlSuccessCallback;
+		brsstrcpy( request.parameters[0].name, "stopmode" );
+		brsstrcpy( request.parameters[0].value, "stop" );  //options - {cycle | instr | stop | qstop} 
+		brsstrcpy( request.parameters[1].name, "usetsp" );
+		brsstrcpy( request.parameters[1].value, "normal" );  //options - {normal | alltsk}
```
Comment (AndrewMusser) [inline src/Ar/A3BR/a3brControl.c:229]: It looks like these were missing for V1 as well, is that the case?

**Review by AndrewMusser — APPROVED** [review]
This all looks really good, nice work sifting through all the API intricacies. I don't think I noticed any changes in this PR to the callbacks, i.e. parsing the return data differently based on V1 vs. V2. Were there no necessary changes there?

File: example/AsProject/Logical/ProcessControl/RobotRwsIntf/Main.st (hunk 2 above)
Comment (JcJakeway) [reply example/AsProject/Logical/ProcessControl/RobotRwsIntf/Main.st:19]: Yes!

File: src/Ar/A3BR/a3brControl.c (hunk 3 above)
Comment (JcJakeway) [reply src/Ar/A3BR/a3brControl.c:229]: It appears to be.  If parameters are not included in the http stop POST, then the default values will be used.  These values of the parameters are set to the default values (stop and normal) so there shouldn't be a change in a robot's expected reaction if it's using v1.0.

**Pull request merged**